src/huntsman/pocs/autoflats.py

```diff
@@ -0,0 +1,428 @@
+import os
+from contextlib import suppress
+from collections import abc, defaultdict
+from concurrent.futures import wait as wait_for_futures
+
+import numpy as np
+from astropy.stats import sigma_clipped_stats
+from astropy import units as u
+from astropy.io import fits
+
+from panoptes.utils.time import current_time, wait_for_events
+from panoptes.utils import get_quantity_value
+from huntsman.pocs.utils.logger import logger as LOGGER
+
+
+class AutoFlatFieldSequence():
+    """ Class to facilitate flat fields with automatic exposure time updates.
+    """
+
+    def __init__(self, cameras, observation, initial_exposure_times=1*u.second,
+                 timeout=10*u.second, min_exptime=0.0001*u.second, max_exptime=60*u.second,
+                 max_attempts=10, required_exposures=5, target_scaling=0.16, scaling_tolerance=0.05,
+                 logger=None, safety_func=None, biases=None):
+        """
+        Args:
```

Comment:
Suggest adding an arg to specify the image slice where statistics are computed. Full frame stats computation is a waste of time and for our strongly vignetted lenses, not really going to be very meaningful.